marshal/sizeCheckUnmarshalizer.go

```diff
@@ -0,0 +1,46 @@
+package marshal
+
+type sizeCheckUnmarshalizer struct {
+	Marshalizer
+	acceptedDelta uint32
+}
+
+// NewSizeCheckUnmarshalizer creates a wrapper around an existing marshalizer m
+// which, during unmarshaling, also checks that the provided buffer dose not contain
+// additional unused data.
+func NewSizeCheckUnmarshalizer(m Marshalizer, maxDelta uint32) Marshalizer {
+	scu := &sizeCheckUnmarshalizer{
+		Marshalizer:   m,
+		acceptedDelta: maxDelta,
+	}
+	return scu
+}
+
+// Unmarshal tries to unmarshal input buffer values into output object, and checks
+// for additional unused data
+func (scu *sizeCheckUnmarshalizer) Unmarshal(obj interface{}, buff []byte) error {
+	err := scu.Marshalizer.Unmarshal(obj, buff)
+	if err != nil {
+		return err
+	}
+
+	out, erro := scu.Marshal(obj)
```

Comment:
This **might** prove to be overkill, needs testing. For all received messages we first unmarshal the bytes and then re-marshal them back. 
Had something similar for a transaction packer and prove to be a bad solution as CPU went most of the time to 100%, go routines started to clutter up and memory consumption increased dramatically. Had to drop that solution and implemented one that was not so precise but it had a lower resource demand. 
If `out of memory` panics will rise in tests, this point is the first to check upon and have a new implementation.
For that, we might make all structures we use for transferring data between nodes implement an interface called `SizeChecker` that is able to roughly compute the memory consumed for the instance as the sum of all contained fields.

Reply:
Ok, let's see how it runs on Irlanda with high transaction load.